linode/manager
# feat: [M3-8099] - Improve Linode Clone Power Off based on initial data

## Description 📝
We'd like to increase the percentage of users powering down their linodes before cloning to reduce the chance of data corruption. We'd also like to capture more analytics data to find out how many users are clicking the new "Power Off" button.

## Changes  🔄
- For both the v1 and v2 Clone flows:
  - Switched the order of the bullet points so that "To help avoid data corruption during the cloning process, we recommend powering off your Compute Instance prior to cloning." comes first in the list.
  - Bolded the text "avoid data corruption".
  - Added an analytics event firing on click of the Power Off button.
  - Users can see their linode's status and power it off in tablet and mobile view in the Linode Clone flow.
  - Added unit test coverage for SelectLinodeCard changes

## Target release date 🗓️
6/10

## Preview 📷

| Before  | After   |
| ------- | ------- |
| ![cloud linode com_linodes_create_linodeID=59147305 regionID=us-sea type=Clone+Linode typeID=g6-nanode-1 (1)](https://github.com/linode/manager/assets/114685994/f725cced-3e34-4441-a8af-ab6708823db1) | ![localhost_3000_linodes_create_type=Backups](https://github.com/linode/manager/assets/114685994/b9d6c4cd-4f2a-41c0-84fd-59913e76383c) |
| ![Screenshot 2024-05-22 at 2 09 53 PM](https://github.com/linode/manager/assets/114685994/262d4ed2-5995-4d50-817d-a8d022905b14) | ![Screenshot 2024-05-22 at 2 10 06 PM](https://github.com/linode/manager/assets/114685994/f374d319-92da-422e-b08a-e19181ba7838) |
| N/A | ![Screenshot 2024-05-22 at 2 05 35 PM](https://github.com/linode/manager/assets/114685994/17e41692-ba4e-419c-a944-22bdf3536f88)|

## How to test 🧪

### Prerequisites
(How to setup test environment)
- Have at least two linodes on your account - one powered on and one powered off.
- Have analytics logging enabled in the browser console. 

### Reproduction steps
(How to reproduce the issue, if applicable)
- Go to https://cloud.linode.com/linodes/create?type=Clone%20Linode.
- Change the screen to mobile view. 
- Observe there is no way to power off a selected, running linode when viewing selection cards.
- Change the screen back to desktop view and select your powered on linode.
- Click the Power Off button and observe in the browser console that there is no analytics event fired when the button is clicked.
- Observe the notice copy has two bullets and may be easy to skim/ignore information about possible corruption.

### Verification steps
(How to verify changes)
- With this PR checked out, go to http://localhost:3000/linodes/create?type=Clone%20Linode.
- Change the screen to mobile view. 
- Confirm that each linode's status is now displaying on the cards.
- Select a powered on linode. Confirm that the selected card shows a Power Off button, and that clicking it powers off the linode and fires an analytics event.
- Change the screen back to desktop view and select your powered on linode.
- Click the Power Off button and confirm in the browser console that is also an analytics event fired when the button is clicked in the table view.
- Observe the notice copy has switched the order of the bullets and bolded text to encourage users to read it.
- Go to http://localhost:3000/linodes/create?type=Backups. (Turn the MSW on if you don't have any backups.) Change the screen to mobile view and confirm that status displays on the cards, but selecting a card of a powered on linode will *not* display the Power Off button.
- Turn on the Linode Create v2 flag in the dev tools and confirm all the above changes are also present for the new version of the create flow.

## As an Author I have considered 🤔

*Check all that apply*
